src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -322,6 +323,61 @@ export function VislibVisualizationsPieChartProvider(Private) {
       }
     }
 
+    readableNumber(number) {
+      let tier = Math.log10(Math.abs(number)) / 3 | 0;
+      if(tier === 0) return number;
+      if(tier > 4) tier = 4;
+      const postfix = this.ABBREVIATIONS[tier];
+      const scale = Math.pow(10, tier * 3);
+      const scaled = number / scale;
+      let formatted = String(scaled.toFixed(1));
+      if (/\.0$/.test(formatted)) {
+        formatted = formatted.substr(0, formatted.length - 2);
+      }
+      return formatted + postfix;
+    }
+
+    addDonutDescriptor(svg, width, height) {
+      const self = this;
```

Comment:
Please don't. Just use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), which will preserve the `this` context.

For new code we don't try to fall back to hacky ES5 solutions anymore.